src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -519,10 +519,20 @@ public SymbolInfo GetSymbolInfo(ExpressionSyntax expression, CancellationToken c
                 // Named arguments handled in special way.
                 return this.GetNamedArgumentSymbolInfo((IdentifierNameSyntax)expression, cancellationToken);
             }
-            else
+            else if (expression.Parent?.Kind() == SyntaxKind.Argument && 
```

Comment:
For correctness, the special case is needed only for implicitly-typed out variable, for other scenarios, existence of BoundTypeExpression would only be an optimization that prevents SemanticModel from binding the type syntax multiple time. Current implementation takes care of the optimization and handles implicitly-typed out variable at the same time. I considered adding BoundTypeExpression node, however, given that it only covers top level type name, adding another field into BoundLocal node, or introducing a new bound node altogether wouldn't pay for itself, I believe.